core/src/main/scala/org/apache/spark/ui/jobs/AllStagesPage.scala

```diff
@@ -19,46 +19,22 @@ package org.apache.spark.ui.jobs
 
 import javax.servlet.http.HttpServletRequest
 
-import scala.xml.{Node, NodeSeq}
+import scala.xml.{Attribute, Elem, Node, NodeSeq, Null, Text}
 
 import org.apache.spark.scheduler.Schedulable
 import org.apache.spark.status.PoolData
-import org.apache.spark.status.api.v1._
+import org.apache.spark.status.api.v1.StageStatus
 import org.apache.spark.ui.{UIUtils, WebUIPage}
 
 /** Page showing list of all ongoing and recently finished stages and pools */
 private[ui] class AllStagesPage(parent: StagesTab) extends WebUIPage("") {
   private val sc = parent.sc
+  private lazy val allStages = parent.store.stageList(null)
```

Comment:
IIRC the class (`AllStagesPage`) is only instantiated once, and the `render` method is called for each request. So this won't really work.